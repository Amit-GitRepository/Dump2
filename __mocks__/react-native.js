var rn = require('react-native');

rn.NativeModules.RNI18n = {languages: ['en']};
rn.NativeModules.BlobModule = {BLOB_URI_SCHEME: ''};
rn.NativeModules.FirebaseDynamicLink = {
  addListener: jest.fn(),
  removeListeners: jest.fn()
};
rn.NativeModules.ReactNativeExceptionHandler = {
  setHandlerforNativeException: jest.fn()
};
jest.mock('AsyncStorage', () => ({
  getItem: jest.fn(() => Promise.resolve('{}')),
  setItem: jest.fn(() => Promise.resolve()),
  mergeItem: jest.fn(() => Promise.resolve()),
  multiGet: jest.fn(() => Promise.resolve('{}'))
}));
jest.mock('BackHandler', () => ({
  addEventListener: jest.fn()
}));
jest.mock('AppState', () => ({
  currentState: 'inactive',
  addEventListener: jest.fn()
}));
var ReactNativeART = require('react-native/Libraries/ART/ReactNativeART');

ReactNativeART.Group = rn.View;
ReactNativeART.Surface = rn.View;
ReactNativeART.Shape = rn.View;
module.exports = rn;
