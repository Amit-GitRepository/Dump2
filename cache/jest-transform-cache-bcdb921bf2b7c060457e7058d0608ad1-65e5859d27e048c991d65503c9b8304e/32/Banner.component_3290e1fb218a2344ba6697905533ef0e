Object.defineProperty(exports,"__esModule",{value:true});var cov_b1u1fqdb3=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/Banner/Banner.component.js',hash='c1a79e7dbe303e3436297b285b03150b4b99cf7a',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/Banner/Banner.component.js',statementMap:{'0':{start:{line:13,column:28},end:{line:13,column:38}},'1':{start:{line:14,column:23},end:{line:14,column:49}},'2':{start:{line:15,column:24},end:{line:15,column:95}},'3':{start:{line:16,column:4},end:{line:16,column:55}},'4':{start:{line:20,column:20},end:{line:20,column:44}},'5':{start:{line:21,column:4},end:{line:21,column:38}},'6':{start:{line:25,column:36},end:{line:25,column:46}},'7':{start:{line:26,column:23},end:{line:26,column:44}},'8':{start:{line:27,column:4},end:{line:31,column:6}},'9':{start:{line:35,column:0},end:{line:39,column:2}},'10':{start:{line:41,column:0},end:{line:46,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:12,column:19},end:{line:12,column:20}},loc:{start:{line:12,column:25},end:{line:17,column:3}},line:12},'1':{name:'(anonymous_1)',decl:{start:{line:19,column:13},end:{line:19,column:14}},loc:{start:{line:19,column:24},end:{line:22,column:3}},line:19},'2':{name:'(anonymous_2)',decl:{start:{line:24,column:2},end:{line:24,column:3}},loc:{start:{line:24,column:12},end:{line:32,column:3}},line:24}},branchMap:{},s:{'0':0,'1':0,'2':0,'3':0,'4':0,'5':0,'6':0,'7':0,'8':0,'9':0,'10':0},f:{'0':0,'1':0,'2':0},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _extends=Object.assign||function(target){for(var i=1;i<arguments.length;i++){var source=arguments[i];for(var key in source){if(Object.prototype.hasOwnProperty.call(source,key)){target[key]=source[key];}}}return target;};var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _resolveAssetSource=require('resolveAssetSource');var _resolveAssetSource2=_interopRequireDefault(_resolveAssetSource);
var _Banner=require('./Banner.styles');var _Banner2=_interopRequireDefault(_Banner);
var _reactNative=require('react-native');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _objectWithoutProperties(obj,keys){var target={};for(var i in obj){if(keys.indexOf(i)>=0)continue;if(!Object.prototype.hasOwnProperty.call(obj,i))continue;target[i]=obj[i];}return target;}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}var

Banner=function(_Component){_inherits(Banner,_Component);function Banner(){var _ref;var _temp,_this,_ret;_classCallCheck(this,Banner);for(var _len=arguments.length,args=Array(_len),_key=0;_key<_len;_key++){args[_key]=arguments[_key];}return _ret=(_temp=(_this=_possibleConstructorReturn(this,(_ref=Banner.__proto__||Object.getPrototypeOf(Banner)).call.apply(_ref,[this].concat(args))),_this),_this.
state={
viewWidth:null},_this.


calculateStyle=function(){cov_b1u1fqdb3.f[0]++;var _ref2=(cov_b1u1fqdb3.s[0]++,
_this.props),source=_ref2.source,style=_ref2.style;
var imageProps=(cov_b1u1fqdb3.s[1]++,(0,_resolveAssetSource2.default)(source));
var heightStyle=(cov_b1u1fqdb3.s[2]++,{height:imageProps.height/imageProps.width*_this.state.viewWidth});cov_b1u1fqdb3.s[3]++;
return _extends({},_Banner2.default.image,heightStyle,style);
},_this.

onLayout=function(event){cov_b1u1fqdb3.f[1]++;var _ref3=(cov_b1u1fqdb3.s[4]++,
event.nativeEvent.layout),width=_ref3.width;cov_b1u1fqdb3.s[5]++;
_this.setState({viewWidth:width});
},_temp),_possibleConstructorReturn(_this,_ret);}_createClass(Banner,[{key:'render',value:function render()

{cov_b1u1fqdb3.f[2]++;var _ref4=(cov_b1u1fqdb3.s[6]++,
this.props),source=_ref4.source,extraProps=_objectWithoutProperties(_ref4,['source']);
var imageStyle=(cov_b1u1fqdb3.s[7]++,this.calculateStyle());cov_b1u1fqdb3.s[8]++;
return(
_react2.default.createElement(_reactNative.ImageBackground,_extends({onLayout:this.onLayout,resizeMode:'contain'},extraProps,{style:imageStyle,source:source}),
this.props.children));


}}]);return Banner;}(_react.Component);cov_b1u1fqdb3.s[9]++;


Banner.defaultProps={
style:{},
source:0,
isBackground:false};cov_b1u1fqdb3.s[10]++;


Banner.propTypes={
style:_propTypes2.default.object,
source:_propTypes2.default.number,
isBackground:_propTypes2.default.bool,
children:_propTypes2.default.node};exports.default=


Banner;