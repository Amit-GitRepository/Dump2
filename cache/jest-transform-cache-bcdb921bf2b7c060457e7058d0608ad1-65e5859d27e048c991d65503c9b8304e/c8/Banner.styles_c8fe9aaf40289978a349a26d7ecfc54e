Object.defineProperty(exports,"__esModule",{value:true});var cov_2rg59obkyk=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/Banner/Banner.styles.js',hash='7eff2c4f0db84b14fca7dc453e947c7ffee37914',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/Banner/Banner.styles.js',statementMap:{},fnMap:{},branchMap:{},s:{},f:{},b:{},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();exports.default={
image:{
width:'100%'}};