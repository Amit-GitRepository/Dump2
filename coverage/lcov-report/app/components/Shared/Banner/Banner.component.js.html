<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/components/Shared/Banner/Banner.component.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">app/components/Shared/Banner</a> Banner.component.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>11/11</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>11/11</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import PropTypes from 'prop-types';
import React, {Component} from 'react';
import resolveAssetSource from 'resolveAssetSource';
import styles from './Banner.styles';
import {ImageBackground} from 'react-native';
&nbsp;
class Banner extends Component {
  state = {
    viewWidth: null
  }
&nbsp;
  calculateStyle = () =&gt; {
    const {source, style} = this.props;
    const imageProps = resolveAssetSource(source);
    const heightStyle = {height: (imageProps.height / imageProps.width) * this.state.viewWidth};
    return {...styles.image, ...heightStyle, ...style};
  }
&nbsp;
  onLayout = (event) =&gt; {
    const {width} = event.nativeEvent.layout;
    this.setState({viewWidth: width});
  }
&nbsp;
  render () {
    const {source, ...extraProps} = this.props;
    const imageStyle = this.calculateStyle();
    return (
      &lt;ImageBackground onLayout={this.onLayout} resizeMode={'contain'} {...extraProps} style={imageStyle} source={source}&gt;
        {this.props.children}
      &lt;/ImageBackground&gt;
    );
  }
}
&nbsp;
Banner.defaultProps = {
  style: {},
  source: 0,
  isBackground: false
};
&nbsp;
Banner.propTypes = {
  style: PropTypes.object,
  source: PropTypes.number,
  isBackground: PropTypes.bool,
  children: PropTypes.node
};
&nbsp;
export default Banner;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Tue Oct 31 2017 11:24:42 GMT+0530 (IST)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
