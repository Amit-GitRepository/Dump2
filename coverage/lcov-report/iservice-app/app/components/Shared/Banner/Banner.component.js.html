<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for iservice-app/app/components/Shared/Banner/Banner.component.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../../index.html">All files</a> / <a href="index.html">iservice-app/app/components/Shared/Banner</a> Banner.component.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">70.59% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>12/17</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">25% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>1/4</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">60% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>3/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">70.59% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>12/17</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">126x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">126x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import PropTypes from 'prop-types';
import React, {Component} from 'react';
import resolveAssetSource from 'resolveAssetSource';
import styles from './Banner.styles';
import {Image, ImageBackground} from 'react-native';
&nbsp;
class Banner extends Component {
  state = {
    viewWidth: null,
    imageStyle: {}
  }
&nbsp;
  calculateStyle = (source, viewWidth, style) =&gt; {
    <span class="missing-if-branch" title="if path not taken" >I</span>if (typeof source === 'object') {
<span class="cstat-no" title="statement not covered" >      Image.getSize(source.uri, <span class="fstat-no" title="function not covered" >(w</span>idth, height) =&gt; {</span>
        const imageStyle = <span class="cstat-no" title="statement not covered" >{height: (height / width) * viewWidth};</span>
<span class="cstat-no" title="statement not covered" >        this.setState({imageStyle: {...styles.image, ...imageStyle, ...style}, viewWidth});</span>
      });
    } else {
      const imageProps = resolveAssetSource(source);
      const imageStyle = {height: (imageProps.height / imageProps.width) * viewWidth};
      this.setState({imageStyle: {...styles.image, ...imageStyle, ...style}, viewWidth});
    }
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  co</span>mponentWillReceiveProps ({source, style}) {
<span class="cstat-no" title="statement not covered" >    if (this.props.source !== source) {</span>
<span class="cstat-no" title="statement not covered" >      this.calculateStyle(source, this.state.viewWidth, style);</span>
    }
  }
&nbsp;
  onLayout = (event) =&gt; {
    const {width} = event.nativeEvent.layout;
    const {style, source} = this.props;
    this.calculateStyle(source, width, style);
  }
&nbsp;
  render () {
    const {source, ...extraProps} = this.props;
    const {imageStyle} = this.state;
    return (
      &lt;ImageBackground onLayout={this.onLayout} resizeMode={'contain'} {...extraProps} getSize={this.calculateStyle} style={imageStyle} source={source}&gt;
        {this.props.children}
      &lt;/ImageBackground&gt;
    );
  }
}
&nbsp;
Banner.defaultProps = {
  style: {},
  source: 0,
  isBackground: false
};
&nbsp;
Banner.propTypes = {
  style: PropTypes.object,
  source: PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.number
  ]),
  isBackground: PropTypes.bool,
  children: PropTypes.node
};
&nbsp;
export default Banner;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sun Dec 03 2017 16:14:06 GMT+0530 (IST)
</div>
</div>
<script src="../../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../../sorter.js"></script>
</body>
</html>
