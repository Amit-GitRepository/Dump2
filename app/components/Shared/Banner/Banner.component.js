import PropTypes from 'prop-types';
import React, {Component} from 'react';
import resolveAssetSource from 'resolveAssetSource';
import styles from './Banner.styles';
import {Image, ImageBackground} from 'react-native';

class Banner extends Component {
  state = {
    viewWidth: null,
    imageStyle: {}
  }

  calculateStyle = (source, viewWidth, style) => {
    if (typeof source === 'object') {
      Image.getSize(source.uri, (width, height) => {
        const imageStyle = {height: (height / width) * viewWidth};
        this.setState({imageStyle: {...styles.image, ...imageStyle, ...style}, viewWidth});
      });
    } else {
      const imageProps = resolveAssetSource(source);
      const imageStyle = {height: (imageProps.height / imageProps.width) * viewWidth};
      this.setState({imageStyle: {...styles.image, ...imageStyle, ...style}, viewWidth});
    }
  }

  componentWillReceiveProps ({source, style}) {
    if (this.props.source !== source) {
      this.calculateStyle(source, this.state.viewWidth, style);
    }
  }

  onLayout = (event) => {
    const {width} = event.nativeEvent.layout;
    const {style, source} = this.props;
    this.calculateStyle(source, width, style);
  }

  render () {
    const {source, ...extraProps} = this.props;
    const {imageStyle} = this.state;
    return (
      <ImageBackground onLayout={this.onLayout} resizeMode={'contain'} {...extraProps} getSize={this.calculateStyle} style={imageStyle} source={source}>
        {this.props.children}
      </ImageBackground>
    );
  }
}

Banner.defaultProps = {
  style: {},
  source: 0,
  isBackground: false
};

Banner.propTypes = {
  style: PropTypes.object,
  source: PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.number
  ]),
  isBackground: PropTypes.bool,
  children: PropTypes.node
};

export default Banner;
