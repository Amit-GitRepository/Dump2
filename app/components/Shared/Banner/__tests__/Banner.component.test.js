import _ from 'lodash';
import Banner from '../Banner.component';
import React from 'react';
import ReactSixteenAdapter from 'enzyme/build/adapters/ReactSixteenAdapter';
import renderer from 'react-test-renderer'; // Enzyme adapter for React 16
import {configure, shallow} from 'enzyme';
import {Text} from 'react-native';

configure({adapter: new ReactSixteenAdapter()});
jest.mock('resolveAssetSource', () => () => ({
  width: 20,
  height: 10
}));
describe('Banner component', () => {
  const wrapper = shallow(<Banner/>);

  it('renders correctly', () => {
    const component = renderer.create(<Banner />).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('renders correctly when isBackground image with children component', () => {
    const component = renderer.create(<Banner isBackground><Text>TEST</Text></Banner>).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('renders correctly', () => {
    expect(wrapper).toBeDefined();
  });

  it('calculatesStyles: the style for imageView', () => {
    wrapper.setState({viewWidth: 30, viewHeight: 30, ratio: 1.5});
    wrapper.instance().calculateStyle(0, 30);
    expect(wrapper.instance().state.imageStyle).toEqual({'height': 15, 'width': '100%'});
  });

  it('onLayout: should set the state with viewWidth on layout', () => {
    const event = _.set({}, 'nativeEvent.layout.width', 10);
    wrapper.instance().onLayout(event);
    expect(wrapper.instance().state.viewWidth).toEqual(10);
  });
});
