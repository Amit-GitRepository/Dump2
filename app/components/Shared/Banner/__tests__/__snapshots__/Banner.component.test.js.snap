// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner component renders correctly 1`] = `
<View
  style={Object {}}
>
  <Image
    getSize={[Function]}
    isBackground={false}
    onLayout={[Function]}
    resizeMode="contain"
    source={0}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
</View>
`;

exports[`Banner component renders correctly when isBackground image with children component 1`] = `
<View
  style={Object {}}
>
  <Image
    getSize={[Function]}
    isBackground={true}
    onLayout={[Function]}
    resizeMode="contain"
    source={0}
    style={
      Array [
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        Object {
          "height": undefined,
          "width": undefined,
        },
        undefined,
      ]
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    TEST
  </Text>
</View>
`;
